main/webapp/modules/core/scripts/views/data-table/cell-renderers/recon-renderer.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * A renderer for cells with recon data.
+ */
+class ReconCellRenderer {
+  render(rowIndex, cellIndex, cell, cellUI) {
+    if (cell && "r" in cell) {
+      var divContent = document.createElement('div');
+      var divContentRecon = $(divContent);
+      var r = cell.r;
+      var service = (r.service) ? ReconciliationManager.getServiceFromUrl(r.service) : null;
+
+      if (r.j == "new") {
+        $('<span>').text(cell.v).appendTo(divContentRecon);
+        $('<span>').addClass("data-table-recon-new").text("new").appendTo(divContentRecon);
+
+        $('<a href="javascript:{}"></a>')
+        .text($.i18n('core-views/choose-match'))
+        .addClass("data-table-recon-action")
+        .appendTo(divContentRecon).on('click',function(evt) {
+          cellUI._doRematch();
+        });
+      } else if (r.j == "matched" && "m" in r && r.m !== null) {
+        var match = cell.r.m;
+        var a = $('<a></a>')
+        .text(match.name)
+        .attr("target", "_blank")
+        .appendTo(divContentRecon);
+
+        if (service && (service.view) && (service.view.url)) {
+          a.attr("href", encodeURI(service.view.url.replace("{{id}}", match.id)));
+        }
+
+        if (DataTableCellUI.previewMatchedCells) {
+          cellUI._previewOnHover(service, match, a, a, false);
```

Comment:
Have you got use cases in mind? I would expect one can already do a lot on the recon service side, no?
I am not opposed to adding an extension point for that as well but it would not be sufficient to implement our thumbnailing use case.